src/sgame/sg_bot_ai.cpp

```diff
@@ -1544,6 +1545,53 @@ static AINodeStatus_t BotActionReachHealA( gentity_t *self )
 	return STATUS_RUNNING;
 }
 
+AINodeStatus_t BotActionExtinguishFire( gentity_t *self, AIGenericNode_t *node )
+{
+	botMemory_t const* mind = self->botMind;
+
+	if ( node != self->botMind->currentNode )
+	{
+		if ( !isBuilder( self ) )
```

Comment:
The granger check should be moved to the beginning of the function, outside the if statement, since it is possible to evolve without changing the current node.

Also it should check for advanced Granger only, not any builder.